WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/Payments Menu/SimplePaymentsMigrationView.swift

```diff
@@ -22,13 +22,17 @@ struct SimplePaymentsMigrationView: View {
             }
             Button(Localization.addCustomAmount) {
                 addCustomAmount()
+                ServiceLocator.analytics.track(event: .SimplePayments.simplePaymentsMigrationSheetAddCustomAmount())
```

Comment:
Comment: Related to the "what if we add an init to this view" question on the https://github.com/woocommerce/woocommerce-ios/pull/12400#discussion_r1560387025. If we do, then we can also pass the analytics instance through `InPersonPaymentsMenuViewModel` from its `InPersonPaymentsMenu` parent view when we initialize this.

Reply:
I like the idea, though I plan to go with `ServiceLocator.analytics` usage instead of DI'ing the analytics to the view because:

- This simple payments migration sheet/modal is supposed to be temporary just as a way to inform merchants about the move to order creation
- The view can't be easily unit-tested: the best approach would be to have a view model for this view `SimplePaymentsMigrationView` with analytics DI'ed for unit tests. However, for the reason above, it might not be worth the effort

Lemme know any other thoughts! 🙂